src/compiler/checker.ts

```diff
@@ -15392,7 +15392,7 @@ namespace ts {
             if (hasSpreadAnyType) {
                 return anyType;
             }
-            return typeToIntersect && spread !== emptyObjectType ? getIntersectionType([typeToIntersect, spread]) : (typeToIntersect || spread);
+            return typeToIntersect && spread !== emptyObjectType ? getIntersectionType([typeToIntersect, spread]) : (typeToIntersect || spread === emptyObjectType ? createJsxAttributesType() : spread);

```

Comment:
can you break this into 3 lines? There might be a more understandable formulation, too, since you're dealing with two related predicates.